pyeudiw/tests/x509/test_x509.py

```diff
@@ -40,25 +39,29 @@ def gen_chain() -> list[bytes]:
         .subject_name(
             x509.Name(
                 [
-                    x509.NameAttribute(NameOID.COMMON_NAME, "ca.example.com"),
+                    x509.NameAttribute(NameOID.COMMON_NAME, ca_cn),
                 ]
             )
         )
         .issuer_name(
             x509.Name(
                 [
-                    x509.NameAttribute(NameOID.COMMON_NAME, "ca.example.com"),
+                    x509.NameAttribute(NameOID.COMMON_NAME, ca_cn),
                 ]
             )
         )
         .public_key(ca_private_key.public_key())
         .serial_number(x509.random_serial_number())
-        .not_valid_before(datetime.utcnow())
-        .not_valid_after(datetime.utcnow() + timedelta(days=365))
+        .not_valid_before(date - timedelta(days=1))
+        .not_valid_after(date + timedelta(days=365))
         .add_extension(
             x509.BasicConstraints(ca=True, path_length=1),
             critical=True,
         )
+        .add_extension(
+            x509.SubjectAlternativeName([x509.DNSName(ca_cn)]),
```

Comment:
DNS name and common name are different

DNS name is a pure hostname, while a CN can be a URI

please align terminology with proper values